craft_parts/plugins/maven_plugin.py

```diff
@@ -152,21 +178,25 @@ def _create_settings(settings_path: Path) -> None:
 
     for protocol in ("http", "https"):
         env_name = f"{protocol}_proxy"
-        if env_name not in os.environ:
+        case_insensitive_env = {item[0].lower(): item[1] for item in os.environ.items()}
+        if env_name not in case_insensitive_env:
             continue
 
-        proxy_url = urlparse(os.environ[env_name])
+        proxy_url = urlparse(case_insensitive_env[env_name])
         proxy = ET.Element("proxy")
         proxy_tags = [
-            ("id", env_name),
+            ("id", env_name.lower()),
             ("active", "true"),
             ("protocol", protocol),
-            ("host", proxy_url.hostname),
+            ("host", str(proxy_url.hostname)),
```

Comment:
how come?

Reply:
Reverted, thank you! (I meant to serialize the port below)